airflow/providers/amazon/aws/utils/waiter_with_logging.py

```diff
@@ -0,0 +1,90 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import logging
+import time
+
+import jmespath
+from botocore.exceptions import WaiterError
+from botocore.waiter import Waiter
+
+from airflow.exceptions import AirflowException
+
+
+def wait(
+    waiter: Waiter,
+    waiter_delay: int,
+    max_attempts: int,
+    state_args: dict,
+    failure_message: str,
+    status_message: dict,
```

Comment:
at first I didn't realize this wasn't just a message... rather than using a dict that's always going to contain the same keys, why not extend this to 2 parameters ? `status_message` could be just the message, like its name suggests, and then have an other param for `status_path` or something.

Reply:
Is there a reason why it would be better to separate the 2 parameters. One benefit of keeping them together is that I can package them into `EMR_SERVERLESS_APPLICATION_STATUS`, which gets used several times. I think this would be the case for multiple operators

Reply:
I just think having the parameters laid out explicitely is more readable and more intuitive to use than having to refer to the doc each time to remember what magic beans I need to put in that bag.
If you want to reuse those, you can use a kwargs-style dict and pass it with `**EMR_SERVERLESS_APPLICATION_STATUS`.
But it's also the opportunity to write a more descriptive status message each time eventually.